simulation/traffic_simulator/test/src/hdmap_utils/test_hdmap_utils.cpp

```diff
@@ -0,0 +1,3152 @@
+static const std::string empty_map_path = "/map/empty/lanelet2_map.osm";
+static const std::string four_track_highway_map_path = "/map/four_track_highway/lanelet2_map.osm";
+static const std::string crossroads_with_stoplines_map_path =
+  "/map/crossroads_with_stoplines/lanelet2_map.osm";
+
+auto makeHdMapUtilsInstance(const std::string relative_path = map_path) -> hdmap_utils::HdMapUtils
+{
+  std::string path =
+    ament_index_cpp::get_package_share_directory("traffic_simulator") + relative_path;
+  geographic_msgs::msg::GeoPoint origin;
+  origin.latitude = 35.61836750154;
+  origin.longitude = 139.78066608243;
+  return hdmap_utils::HdMapUtils(path, origin);
+}
+
+auto makePoint(const double x, const double y, const double z = 0.0) -> geometry_msgs::msg::Point
+{
+  geometry_msgs::msg::Point point;
+  point.x = x;
+  point.y = y;
+  point.z = z;
+  return point;
+}
+
+auto makeBoundingBox(const double center_y = 0.0) -> traffic_simulator_msgs::msg::BoundingBox
+{
+  traffic_simulator_msgs::msg::BoundingBox bbox;
+  bbox.center.x = 1.0;
+  bbox.center.y = center_y;
+  bbox.dimensions.x = 4.0;
+  bbox.dimensions.y = 2.0;
+  bbox.dimensions.z = 1.5;
+  return bbox;
+}
+
+auto makePose(
+  geometry_msgs::msg::Point position,
+  geometry_msgs::msg::Quaternion orientation = geometry_msgs::msg::Quaternion())
+  -> geometry_msgs::msg::Pose
+{
+  geometry_msgs::msg::Pose pose;
+  pose.position = position;
+  pose.orientation = orientation;
+  return pose;
+}
+
+auto makeSmallBoundingBox(const double center_y = 0.0) -> traffic_simulator_msgs::msg::BoundingBox
+{
+  traffic_simulator_msgs::msg::BoundingBox bbox;
+  bbox.center.x = 0.0;
+  bbox.center.y = center_y;
+  bbox.dimensions.x = 1.0;
+  bbox.dimensions.y = 1.0;
+  bbox.dimensions.z = 1.0;
+  return bbox;
+}
+
+auto makeQuaternionFromYaw(const double yaw) -> geometry_msgs::msg::Quaternion
+{
+  geometry_msgs::msg::Vector3 v;
+  v.z = yaw;
+  return quaternion_operation::convertEulerAngleToQuaternion(v);
+}
+
+int main(int argc, char ** argv)
+{
+  testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
+
+/**
+ * @note Test basic functionality; test initialization correctness with a correct path to a lanelet map.
+ */
+TEST(HdMapUtils, Construct) { ASSERT_NO_THROW(auto hdmap_utils = makeHdMapUtilsInstance()); }
+
+/**
+ * @note Test basic functionality; test initialization correctness with an invalid path to a lanelet map.
+ */
+TEST(HdMapUtils, Construct_invalid)
+{
+  std::string path =
+    ament_index_cpp::get_package_share_directory("traffic_simulator") + "invalid_path";
+  geographic_msgs::msg::GeoPoint origin;
+  origin.latitude = 35.61836750154;
+  origin.longitude = 139.78066608243;
+  EXPECT_THROW(hdmap_utils::HdMapUtils hdmap_utils(path, origin), std::runtime_error);
+}
+
+/**
+ * @note Test basic functionality; test map conversion to binary message correctness with a sample map.
+ */
+TEST(HdMapUtils, toMapBin)
+{
+  auto hdmap_utils = makeHdMapUtilsInstance();
```

Comment:
Almost all tests create an instance of HdMapUtils, but this item can be reduced in description by using Gtest's [TestFixture](https://google.github.io/googletest/primer.html#same-data-multiple-tests). Please use it to reduce the amount of description.